lib/spack/spack/environment/environment.py

```diff
@@ -135,7 +135,7 @@ def default_manifest_yaml():
 valid_environment_name_re = r"^\w[\w-]*$"
 
 #: version of the lockfile format. Must increase monotonically.
-lockfile_format_version = 5
+lockfile_format_version = 6
```

Comment:
update docs too -- the lockfile docs go through the different spec schemas: https://spack.readthedocs.io/en/latest/spack.environment.html#spack-lock-format

Reply:
See https://github.com/spack/spack/pull/45189/commits/1aa74bd6761ce18f128214957c723ae04c09037d